detect-secrets.sh

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env bash
+
+#
+# Copyright contributors to the Galasa project
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+#-----------------------------------------------------------------------------------------
+#
+# Objectives: Detect secrets in every repo in galasa, this will prevent commiting any
+#             secrets to Github. If it finds any secrets, build should fail!
+#
+#-----------------------------------------------------------------------------------------
+
+# Where is this script executing from ?
+BASEDIR=$(dirname "$0")
+pushd $BASEDIR 2>&1 >>/dev/null
+BASEDIR=$(pwd)
+popd 2>&1 >>/dev/null
+# echo "Running from directory ${BASEDIR}"
+export ORIGINAL_DIR=$(pwd)
+# cd "${BASEDIR}"
+
+cd "${BASEDIR}/.."
+REPO_ROOT=$(pwd)
+
+#-----------------------------------------------------------------------------------------
+#
+# Set Colors
+#
+#-----------------------------------------------------------------------------------------
+bold=$(tput bold)
+underline=$(tput sgr 0 1)
+reset=$(tput sgr0)
+red=$(tput setaf 1)
+green=$(tput setaf 76)
+white=$(tput setaf 7)
+tan=$(tput setaf 202)
+blue=$(tput setaf 25)
+
+#-----------------------------------------------------------------------------------------
+#
+# Headers and Logging
+#
+#-----------------------------------------------------------------------------------------
+underline() { printf "${underline}${bold}%s${reset}\n" "$@"; }
+h1() { printf "\n${underline}${bold}${blue}%s${reset}\n" "$@"; }
+h2() { printf "\n${underline}${bold}${white}%s${reset}\n" "$@"; }
+debug() { printf "${white}[.] %s${reset}\n" "$@"; }
+info() { printf "${white}[➜] %s${reset}\n" "$@"; }
+success() { printf "${white}[${green}✔${white}] ${green}%s${reset}\n" "$@"; }
+error() { printf "${white}[${red}✖${white}] ${red}%s${reset}\n" "$@"; }
+warn() { printf "${white}[${tan}➜${white}] ${tan}%s${reset}\n" "$@"; }
+bold() { printf "${bold}%s${reset}\n" "$@"; }
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@"; }
+
+#-----------------------------------------------------------------------------------------
+# Functions
+#-----------------------------------------------------------------------------------------
+
+function usage {
+    info "Syntax: detect-secrets.sh [OPTIONS]"
+    cat <<EOF
+Options are:
+-h | --help : Display this help text
+--module The name of the module to start building from
+--chain true/false/yes/no/y/n
```

Comment:
This script doesn't take these flags so these should be removed from this help text.